src/KubernetesClient/WatcherDelegatingHandler.cs

```diff
@@ -27,20 +27,63 @@ internal class WatcherDelegatingHandler : DelegatingHandler
 
                 if (query.TryGetValue("watch", out var values) && values.Any(v => v == "true"))
                 {
-                    originResponse.Content = new LineSeparatedHttpContent(originResponse.Content);
+                    originResponse.Content = new LineSeparatedHttpContent(originResponse.Content, cancellationToken);
                 }
             }
+
             return originResponse;
         }
 
+        internal class CancelableStream : Stream
+        {
+            private readonly Stream _innerStream;
+            private readonly CancellationToken _cancellationToken;
+
+            public CancelableStream(Stream innerStream, CancellationToken cancellationToken)
+            {
+                _innerStream = innerStream;
+                _cancellationToken = cancellationToken;
+            }
+
+            public override void Flush() => _innerStream.Flush();
+
+            public override int Read(byte[] buffer, int offset, int count) =>
+                _innerStream.ReadAsync(buffer, offset, count, _cancellationToken).ConfigureAwait(false)
```

Comment:
Consider omitting `ConfigureAwait(false)` as it does not affect the behaviour of the awaiter's `GetResult`. It is also potentially misleading to the reader as they may falsely believe these methods (`CancelableStream.{Read, Write}`) to be insusceptible to the classic async/await deadlock.